contract/src/main/java/io/neow3j/contract/Invocation.java

```diff
@@ -156,6 +159,50 @@ public void addWitnesses(Witness... witnesses) {
         }
     }
 
+    /**
+     * Checks if the sender account of this invocation can cover the network and system fees. If
+     * not, executes the given consumer supplying it with the required fee and the sender's GAS
+     * balance.
+     *
+     * @return this.
+     * @throws IOException if something goes wrong in the communication with the neo-node.
+     */
+    public Invocation doIfSenderCannotCoverFees(BiConsumer<BigInteger, BigInteger> consumer)
+            throws IOException {
+        BigInteger fees = BigInteger.valueOf(
+                this.transaction.getSystemFee() + this.transaction.getNetworkFee());
+        BigInteger senderGasBalance = new GasToken(this.neow)
+                .getBalanceOf(this.transaction.getSender());
+        if (fees.compareTo(senderGasBalance) < 0) {
```

Comment:
Couldn't extract method from this because the fee and balance values are needed for the consumer below.